### PR TITLE
Optimization for Port creation.

### PR DESCRIPTION
**Pull request by zzxgzgz**
This PR is made for the purpose of optimizing the time needed to create Ports. 

Changes include:
1. Changed the way of inserting Ports and Hosts to database, from inserting one by one, to inserting in a bulk. 
2. In one `/ports POST` request, the manager doesn't need to query for the hosts all the time, instead we use a `set()` to store the Hosts to add in this call, and look into it when creating Ports. 

Observed results:
For the same size of data(100k in total, 1k per batch), the zeta controller used to take about 3.2s to create 1k ports; with this change, the time is dropped to about 1.6s for 1k ports.

### CODE REVIEW

**Comment by liangbin-pub** [comment]
Once the data path change is in(issue #36), please resolve possible conflicts and update the PR, thanks!

File: src/mgmt/manager/project/api/ports.py [hunk 1]
```diff
@@ -46,48 +46,72 @@ def ports_update_eps_config():
...
+            if (host is None) and (ip_node_to_find not in host_ip_node_lookup_set):
+                host_to_add = {'mac_node': post_data.get('mac_node'),
+                    'ip_node' : ip_node_to_find,
+                    'host_id' : str(uuid.uuid4())}
+                port['host_id'] = host_to_add['host_id']
+                hosts_to_add.append(host_to_add)
+                host_ip_node_lookup_set.add(ip_node_to_find)
+            elif (host is not None):
+                port['host_id'] = host.host_id
+            elif (ip_node_to_find in host_ip_node_lookup_set):
```
Comment (liangbin-pub) [inline src/mgmt/manager/project/api/ports.py]: There is no other possibilities, so this should be just "else"

**Review by liangbin-pub — APPROVED** [review]
Looks good.
Please make sure sync with zeta main after PR #67 merged and add data path ping pass result here before merge

**Comment by liangbin-pub** [comment]
@zzxgzgz Let's resolve the comment and verify data path then have this merged. This one is the last piece of rel v0.1.0

File: src/mgmt/manager/project/api/ports.py (hunk 1 above)
Comment (zzxgzgz) [reply src/mgmt/manager/project/api/ports.py]: fixed

**Review by liangbin-pub — APPROVED** [review]
Looks good

**Comment by zzxgzgz** [comment]
image
Ping was successful

**Pull request merged**